src/openms/source/ANALYSIS/OPENSWATH/TransitionPQPFile.cpp

```diff
@@ -670,6 +670,43 @@ namespace OpenMS
   }
 
   // public methods
+  std::unordered_map<std::string, std::string> TransitionPQPFile::getPQPIDToTraMLIDMap(const char* filename, std::string tableName)
+  {
+    sqlite3 *db;
```

Comment:
using Sqlitecpp with
https://github.com/OpenMS/OpenMS/blob/d5f97e9c5c20edf0996e3b58fc7ba703e9232d74/src/openms/source/FORMAT/OMSFileLoad.cpp#L46
would ensure that memory is properly freed

Reply:
but not a must

Reply:
Sorry I don't quite understand what you mean by this. Would I have to incorporate that line into the method signature? I don't think TransitionPQPFile currently uses a db_ variable though

Reply:
I modelled the db handling after `readPQPInput_` so I am not sure if that has to be changed too. https://github.com/OpenMS/OpenMS/blob/63090facde39834d6e2646a810d6c3178e4b65f7/src/openms/source/ANALYSIS/OPENSWATH/TransitionPQPFile.cpp#L33 


It seems that the database is closed when the object is deleted. Not sure if this happens automatically at the end of the function call or it has to be invoked manually. https://github.com/OpenMS/OpenMS/blob/483d1ee831a5a9b49dfaa1210f148c81225e6fe5/src/openms/source/FORMAT/SqliteConnector.cpp#L24-L31